apps/devhub/tests/test_views.py

```diff
@@ -2723,6 +2743,24 @@ def test_unlisted_addon_pass_validation_with_flag(self, mock_sign_file):
         assert file_.status == amo.STATUS_LITE
         assert mock_sign_file.called
 
+    @mock.patch('devhub.views.sign_file')
+    def test_listed_beta_pass_validation_with_flag(self, mock_sign_file):
+        """Beta files that pass validation are signed with prelim cert if flag
+        is enabled."""
+        self.create_flag('automatic-validation')
+        self.addon.update(
+            is_listed=True, status=amo.STATUS_PUBLIC, trusted=False)
+        # Make sure the file has no validation warnings nor errors.
+        self.upload.update(
+            validation='{"notices": 2, "errors": 0, "messages": [],'
+                       ' "metadata": {}, "warnings": 0}')
+        assert self.addon.status == amo.STATUS_PUBLIC  # Fully reviewed.
+        self.post(beta=True)
+        file_ = File.objects.all().order_by("-created")[0]
+        assert self.addon.status == amo.STATUS_PUBLIC
+        assert file_.status == amo.STATUS_BETA
+        assert mock_sign_file.called
```

Comment:
@magopian I'm not sure where this test should go. Under `TestAddVersion` or `TestVersionAddFile`? I'll get rid of one of the above.